pkg/trie/triedb/triedb.go

```diff
@@ -358,15 +359,16 @@ type inspectResult struct {
 	changed bool
 }
 
-// inspect inspects the given node `stored` and calls the `inspector` function
+// inspect inspects the given node stored and calls the inspector function
 // then returns the new node and a boolean indicating if the node has changed
 func (t *TrieDB[H, Hasher]) inspect(
 	stored StoredNode,
 	key *nibbles.Nibbles,
 	inspector func(Node, *nibbles.Nibbles) (action, error),
 ) (*inspectResult, error) {
 	// shallow copy since key will change offset through inspector
```

Comment:
```suggestion
```
assuming `key.Clone()` does not return a shallow copy

Reply:
from the `slices.Clone` functionality, they use assignment so I would consider it a shallow clone.  The actual offset is copied via assignment as well so I would consider that as a shallow copy.